components/private/Snapshot/MDCSnapshotTestCase.m

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCSnapshotTestCase.h"
+
+@implementation MDCSnapshotTestCase
+
+- (void)setUp {
+  [super setUp];
+  self.agnosticOptions = FBSnapshotTestCaseAgnosticOptionOS;
+}
+
+- (UIView *)addBackgroundViewToView:(UIView *)view {
+  UIView *backgroundView =
+      [[UIView alloc] initWithFrame:CGRectMake(0, 0, CGRectGetWidth(view.bounds) + 20,
+                                               CGRectGetHeight(view.bounds) + 20)];
+  backgroundView.backgroundColor = [UIColor colorWithWhite:0.8 alpha:1];
+  [backgroundView addSubview:view];
+  view.center = backgroundView.center;
+  return backgroundView;
+}
+
+- (void)snapshotVerifyView:(UIView *)view {
+  // TODO(https://github.com/material-components/material-components-ios/issues/5888)
+  // Support multiple OS versions for snapshots
+  if (NSProcessInfo.processInfo.operatingSystemVersion.majorVersion != 11 ||
+      NSProcessInfo.processInfo.operatingSystemVersion.minorVersion != 2 ||
+      NSProcessInfo.processInfo.operatingSystemVersion.patchVersion != 0) {
+    NSLog(@"Skipping this test. Snapshot tests currently only run on iOS 11.2.0");
+    return;
+  }
+
+  UIImage *result = nil;
+
```

Comment:
It's a little confusing to follow all logic paths for the different versions in this method. Anyway to make it a little clearer?

Reply:
Acknowledged. This function could be refactored/cleaned up as part of a later PR.